src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBase.cpp

```diff
@@ -397,6 +398,64 @@ void CompositionalMultiphaseBase::initializePreSubGroups()
   initializeAquiferBC( cm );
 }
 
+void CompositionalMultiphaseBase::validateConstitutiveModels() const
+{
+  GEOSX_MARK_FUNCTION;
+
+  DomainPartition const & domain = this->getGroupByPath< DomainPartition >( "/Problem/domain" );
```

Comment:
While we do this in quite a few places now, I think it's best to limit this access pattern when we can, to make replacing it easier if we ever decide to. Here the function is always called from `initializePreSubGroups()` which already extracts the domain in this way, so we should just pass it as a parameter.

Reply:
Yes, I agree. I now pass the `DomainPartition` as a parameter to the validation function